src/mpid/common/genq/mpidu_genq_shmem_types.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#ifndef MPIDU_GENQI_SHMEM_TYPES_H_INCLUDED
+#define MPIDU_GENQI_SHMEM_TYPES_H_INCLUDED
+
+#include "mpl.h"
+#include "mpidu_init_shm.h"
+
+/* Public Interface */
+
+typedef void *MPIDU_genq_shmem_pool_t;
+
+typedef struct MPIDU_genq_shmem_queues_t {
+    void *base;                 /* base address for the cells */
+    void *ptr;                  /* pointer to shmem allocation */
+} MPIDU_genq_shmem_queues_t;
```

Comment:
Essentially, this is creating a local array of reference to the shared queue object. The purpose seems to be avoiding passing pool as a parameter for enqueue/dequeue operation?

Reply:
You don't need pool for enqueue/dequeue, you only need base address.